tools/screenshots.sh

```diff
@@ -0,0 +1,748 @@
+    command -v wget >/dev/null 2>&1 || { show_error_message "wget command not installed. Please, install it!"; stop_on_error; }
+    wget $FONT_ZIP_URL -O "$WORKING_DIR/noto.zip" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message "Done"
+
+    show_message "Unzipping..."
+    unzip "$WORKING_DIR/noto.zip" -d "$WORKING_DIR/$FONT_DIR/" >> $LOG_FILE 2>&1 || stop_on_error
+    show_message "Done"
+  fi
+}
+
+# Checks for ImageMagick
+function require_imagemagick {
+  which magick &>/dev/null
+  immissing=$?
+
+  if [ $immissing = 1 ]; then
+    show_message "Installing ImageMagick..."
+    exec brew install imagemagick
+  fi
+}
+
+# Setups the local sdk and configure the environment
+function require_sdk() {
+  if [ -z "$ANDROID_SDK_DIR" ]; then
+    require_dirs
+
+    if [ ! -d "$WORKING_DIR/sdk/tools" ]; then
+      command -v wget >/dev/null 2>&1 || { show_message "wget command not installed"; stop_on_error; }
+
+      show_message "Downloading the Android SDK..."
+      mkdir -p "$WORKING_DIR/sdk" >> $LOG_FILE 2>&1 || stop_on_error
+      wget -qO- $SDK_ZIP_URL | tar xf - -C "$WORKING_DIR/sdk" >> $LOG_FILE 2>&1 || stop_on_error
+      show_message Done
+
+      show_message "Downloading the Android platform (over 2GB so, this might take a while)"
+      yes y | sdkmanager "platform-tools" "platforms;android-26" "system-images;android-25;google_apis;x86" "emulator" >> $LOG_FILE 2>&1 || stop_on_error
+      show_message Done
+    fi
+  fi
+
+  show_message "Setting up SDK..."
+  export ANDROID_SDK_DIR="$WORKING_DIR/sdk"
+  export PATH=$PATH:"$WORKING_DIR/sdk/tools":"$WORKING_DIR/sdk/tools/bin":"$WORKING_DIR/sdk/platform-tools/"
+  show_message "Done"
+}
+
+# Creates the local emulator
+function require_emu {
+  avdmanager list avd -c | grep $AVD
+  avdmissing=$?
+
+  if [ $avdmissing = 1 ]; then
+    show_message "Creating AVD..."
+    echo no | avdmanager create avd -n $AVD -k "system-images;android-25;google_apis;x86" --tag "google_apis" >> $LOG_FILE 2>&1 || stop_on_error
+  fi
+}
+
+# Setups the environment
+function execute_setup() {
+  show_title_message "Setting up the emulation environment..."
+  require_dirs
+
+  needSdk=0
+  command -v avdmanager >/dev/null 2>&1 || needSdk=1
+  command -v adb >/dev/null 2>&1 || needSdk=1
+  command -v emulator >/dev/null 2>&1 || needSdk=1
+
+  if [ $needSdk -eq 1 ]; then
+    require_sdk
+  fi
+  require_emu
+  show_ok_message "Done!"
+}
+
+# Loads and checks the token for the magic login
+# Also loads other configuration that is in the screenshot-config.sh file
+function require_deeplink {
+  if [ -f "screenshots-config.sh" ]; then
+    . screenshots-config.sh
+  fi
+
+  if [ -z "$TOKEN_DEEPLINK" ]; then
+    show_error_message "TOKEN_DEEPLINK variable is not set correctly. Make sure the file screenshots-config.sh is present and looks like this:"
+    show_error_message ""
+    show_error_message "#!/bin/sh"
+    show_error_message "TOKEN_DEEPLINK=wordpress://magic-login?token=<secret login token>" 
```

Comment:
Can we use a simpler username/password combination instead of a magic link token? Also, IIRC this token is temporary and works for 1 hour before it expires. Can be a bit unpractical if the release manager has to update screenshots regularly.

Reply:
I was told it's possible to obtain token with longer expire time :-(
Anyway, I like the user/password solution more... so let's go with that.